cmd/remote-storage/app/config.go

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2025 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package app
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/spf13/viper"
+	"go.opentelemetry.io/collector/config/configgrpc"
+	"go.opentelemetry.io/collector/config/confignet"
+	"go.opentelemetry.io/collector/confmap"
+
+	cascfg "github.com/jaegertracing/jaeger/internal/storage/cassandra/config"
+	escfg "github.com/jaegertracing/jaeger/internal/storage/elasticsearch/config"
+	"github.com/jaegertracing/jaeger/internal/storage/v1/badger"
+	"github.com/jaegertracing/jaeger/internal/storage/v1/cassandra"
+	es "github.com/jaegertracing/jaeger/internal/storage/v1/elasticsearch"
+	"github.com/jaegertracing/jaeger/internal/storage/v1/memory"
+	"github.com/jaegertracing/jaeger/internal/storage/v2/clickhouse"
+	"github.com/jaegertracing/jaeger/internal/storage/v2/grpc"
+	"github.com/jaegertracing/jaeger/internal/tenancy"
+)
+
+// StorageConfig contains configuration for storage backends.
+// This is based on jaegerstorage.Config but placed here to avoid internal package imports.
+type StorageConfig struct {
+	Backends map[string]TraceBackend `mapstructure:"backends"`
+}
+
+// TraceBackend contains configuration for a single trace storage backend.
+// This mirrors jaegerstorage.TraceBackend.
+type TraceBackend struct {
```

Comment:
I don't like the duplication between remote-storage and the storage_extension in the jaeger binary. I suggest we move the storage configuration portion into cmd/internal/storageconfig. Basically the code from this point https://github.com/jaegertracing/jaeger/blob/cb37388d294f2b249a0736129c100d44199a71b0/cmd/jaeger/internal/extension/jaegerstorage/extension.go#L161 which instantiates trace storage and metrics storage factories should move to a shared location